src/main/java/org/folio/fqm/service/FqlToSqlConverterService.java

```diff
@@ -271,16 +296,19 @@ private static Condition handleNotContainsAny(NotContainsAnyCondition notContain
   }
 
   private static Condition handleEmpty(EmptyCondition emptyCondition, EntityType entityType, org.jooq.Field<Object> field) {
-    String fieldType = getFieldForFiltering(emptyCondition, entityType)
-      .getDataType()
-      .getDataType();
+    String fieldType = getFieldDataType(entityType, emptyCondition);
     boolean isEmpty = Boolean.TRUE.equals(emptyCondition.value());
     var nullCondition = isEmpty ? field.isNull() : field.isNotNull();
+
     return switch (fieldType) {
       case STRING_TYPE -> isEmpty ? nullCondition.or(field.eq("")) : nullCondition.and(field.ne(""));
       case "arrayType" -> {
-        var cardinality = cardinality(DSL.cast(field, String[].class));
-        yield isEmpty ? nullCondition.or(cardinality.eq(0)) :  nullCondition.and(cardinality.ne(0));
+        var cardinality = cardinality(cast(field, String[].class));
+        if (isEmpty) {
+          yield nullCondition.or(cardinality.eq(0)).or(ALL_NULLS.formatted(field));
```

Comment:
Rather than pass values in with `.formatted()`, can we parameterize this? If not, we risk opening up a pretty big SQL injection hole

Reply:
I guess it's already partially parameterized by `or()`, so there's probably not really a security concern here

Reply:
I totally understand the concern, that's the main reason I didn't really like this approach. Since the entire condition is being passed as a string, I couldn't figure out how to parameterize it in this context (since we want to pass the whole argument as a string, but just want the field name to be parameterized). But like you said, I think it should be ok with the `or()`, plus the fact that only our predefined valueGetters/filterValueGetters will ever be used here.